### PR TITLE
Add breach resolution feature - WIP 

### PR DESCRIPTION
**Pull request by lesleyjanenorton**
WIP breach resolution feature updates so @groovecoder and I can get our Resolved Breach Ducks in a row. 🦆🦆🦆

🦆 `/dashboard` Updates: 
- If at least one breach has been resolved, the second and third "breach stats" at page top change to reflect the number of resolved passwords and the number of passwords still exposed in unresolved breaches.
- Previously resolved breaches are hidden by default and gated behind "Show Resolved" buttons.
- The CTA for unresolved breach cards in the dashboard is changed to "Resolve this breach"
- The CTA for resolved breach cards reverts back to "More about this breach"

Adds a progress bar to the dashboard and...
- If a user has at least one breach, an intro to resolving breaches is visible below the breach stats.
- If a user has resolved at least one breach, the intro is replaced with a progress bar that  indicates the number of resolved/unresolved breaches.
- If a user has resolved all of their breaches, the progress bar is replaced with some messaging telling the user they've resolved everything.

🦆 `/breach-details` Updates: 
- Adds a top of page notification if the user has an email that was exposed in that breach and the breach hasn't been marked as resolved.
- Adds list of all verified emails that were exposed in the breach and adds ability to mark them as resolved/unresolved
- Adds a confirmation modal with varying messaging and imagery based on the number of resolved breaches.

### CODE REVIEW

File: template-helpers/breach-stats.js [hunk 1]
```diff
@@ -1,21 +1,110 @@
...
+    // If a user has resolved at least one breach:
+    // Change the password stat to show the number of password-exposing unresolved breaches.
+    const remainingExposedPasswords = passwordStat.count - passwordStat.numResolved;
+    passwordStat.subhead = localize(locales, "unresolved-passwords-exposed", { numPasswords: remainingExposedPasswords });
+    passwordStat.displayCount = remainingExposedPasswords;
+
+    // Change the total number of breaches callout to show the total number of resolved breaches
+    breachesStat.subhead = localize(locales, "known-data-breaches-resolved", { numBreaches: breachesStat.numResolved });
+    breachesStat.displayCount = breachesStat.numResolved;
+  } else {
```
Comment (groovecoder) [inline template-helpers/breach-stats.js]: Can we return early and drop the `else` block?

**Comment by flodolo** [comment]
I'm going to mute notifications, since this doesn't seem to have FTL anymore. Yell really loud if you need me ;-)

File: controllers/user.js [hunk 2]
```diff
@@ -309,6 +309,74 @@ async function postRemoveFxm(req, res) {
...
 
+// Placeholder -- WIP
+async function postResolveBreach(req, res) {
+  const user = req.user;
+
+  // Currently we're sending { affectedEmail, recencyIndex, isResolved, passwordsExposed } in req.body
+  // Not sure if we need all of these or need to send other/additional values?
+
+  const { isResolved } = req.body;
+  if (isResolved === "true") {
```
[the remark this answers is not in the document]
Comment (lesleyjanenorton) [reply controllers/user.js]: Confirmed

File: controllers/breach-details.js [hunk 3]
```diff
@@ -4,6 +4,28 @@ const HIBP = require("../hibp");
...
+
+        // affectedEmails = [
+        //   {
+        //     emailAddress: "firstInvolvedEmail@userEmail.com",
+        //     recencyIndex: "1",
+        //     isResolved: false,
+        //   },
+        //   {
+        //     emailAddress: "secondInvolvedEmail@mailinator.com",
+        //     recencyIndex: "4",
```
[the remark this answers is not in the document]
Comment (lesleyjanenorton) [reply controllers/breach-details.js]: Oh good point- probably numbers? @groovecoder are we storing these as numbers or strings?

File: controllers/user.js [hunk 4]
```diff
@@ -309,6 +309,74 @@ async function postRemoveFxm(req, res) {
...
+
+  if (numResolvedBreaches === 2) {
+    localizedModalStrings.headline = req.fluentFormat("confirmation-2-subhead");
+    localizedModalStrings.progressMessage = req.fluentFormat("confirmation-2-body");
+    localizedModalStrings.headlineClassName = "overlay-take-that-hackers";
+  }
+
+  if (numResolvedBreaches === 3) {
+    localizedModalStrings.headline = req.fluentFormat("confirmation-3-subhead");
+    // TO CONSONDER: The "confirmation-3-body" string contains nested markup.
```
[the remark this answers is not in the document]
Comment (lesleyjanenorton) [reply controllers/user.js]: 🤦‍♀️yes

**Comment by groovecoder** [comment]
Closing in favor of https://github.com/mozilla/blurts-server/pull/1481 which is re-based.

**Pull request closed without merging**